librz/reg/arena.c

```diff
@@ -168,15 +168,19 @@ RZ_API int rz_reg_fit_arena(RzReg *reg) {
 
 RZ_API RzRegArena *rz_reg_arena_new(int size) {
 	RzRegArena *arena = RZ_NEW0 (RzRegArena);
-	if (arena) {
-		if (size < 1) {
-			size = 1;
-		}
-		if (!(arena->bytes = calloc (1, size + 8))) {
-			RZ_FREE (arena);
-		} else {
-			arena->size = size;
-		}
+	if (!arena) {
+		RZ_LOG_ERROR ("Failed to allocate RzRegArena.\n");
+		return NULL;
+	}
+
+	arena->size = size;
+	if (size < 1) {
+		return arena;
```

Comment:
i would suggest to also free and return NULL here.

Reply:
This is not an error. It's a valid use case, but if size is 0 (`size` should be of type `size_t`, to make sure it is always positive or 0) you don't want to allocate `arena->bytes` for nothing.